### DISCUSSION TITLE
How to download files from external host

### DISCUSSION THREAD

**Discussion by higoka** [sindresorhus/ky#597] [Q&A] [answered]
I have a list of filenames i need to download from a website.

In nodejs how would i download these files in parallel using `ky`? is this even possible or should i use `got`?

**Comment by sindresorhus** [comment, marked as answer]
Something like this (untested):

```js
import ky from 'ky';
import {writeFile} from 'fs/promises';

async function downloadFiles(fileNames, downloadUrl) {
	await Promise.all(
		fileNames.map(async fileName => {
			const file = await ky.get(`${downloadUrl}/${fileName}`).bytes();
			await writeFile(fileName, file);
		})
	);
}

const fileNames = ['file1.txt', 'file2.txt']; // Replace with your filenames
const downloadUrl = 'https://example.com/files'; // Replace with your download URL
await downloadFiles(fileNames, downloadUrl);
```

**Reply by raphaelcockx** [reply]
I just tried (a variation on) this code but noticed that you can't just pass `file` to `writeFile` as it doesn't accept an arrayBuffer for the second argument. This however will work:

```js
await writeFile(fileName, Buffer.from(file));
```

**Reply by sindresorhus** [reply]
If you are on Node.js 20 or later, you can use `.bytes()` instead of `.arrayBuffer()`.
